### PR TITLE
Push notifications: Add Firebase

### PR DESCRIPTION
**Pull request by noahmalmed**
First part to https://github.com/zooniverse/mobile/projects/2#card-9508797.

Still TODO: 
- Update Settings
- Update README
- Create documentation on sending push notifications

## Description: 
This pull request rips out pusher and integrates Firebase cloud messaging as our platform. The reason I decided to make the switch is because pusher is moving out of Beta soon and would require a paid plan outside of the one we are paying for. Also, we don't really need all of the features Pusher provides.

Firebase Cloud messaging is free and integrates with both Android and iOS and is fairly intuitive to use.

## Details
- Remove Pusher
- Add Firebase Cloud Messaging (FCM) Dependencies
- Move user data to redux persist instead of using the simple store

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

### CODE REVIEW

**Comment by wgranger** [comment]
If we are removing Pusher _completely_ there are still a few mentions of pusher in line 24 and 25 of the `README` as well as line 71 of `zooniverseApp.js`. There is also a mention of the `PUSHER_API_KEY` in the build.gradle.

File: src/containers/zooniverseApp.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ class ZooniverseApp extends Component {
   }
 
   onPushRegistration = () => {
-    this.props.syncInterestSubscriptions()
+    // this.props.syncInterestSubscriptions()
```
Comment (wgranger) [inline src/containers/zooniverseApp.js]: Is this being saved for later? Seems unnecessary.

File: src/components/ProjectNotification.js [hunk 2]
```diff
@@ -16,9 +16,9 @@ const mapStateToProps = (state, ownProps) => ({
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
   updateProjectNotification(checked) {
-    dispatch(setState(`notifications.${ownProps.id}`, checked))
-    dispatch(syncNotificationStore())
-    dispatch(updateInterestSubscription(ownProps.id, checked))
+    // dispatch(setState(`notifications.${ownProps.id}`, checked))
+    // dispatch(syncNotificationStore())
+    // dispatch(updateInterestSubscription(ownProps.id, checked))
```
Comment (wgranger) [inline src/components/ProjectNotification.js]: Similar comment as before, is there a reason we are keeping this around, commented out?

**Comment by noahmalmed** [comment]
Login to https://console.firebase.google.com as android@zooniverse.com (password is in pass bolt).

From there, go to the `zooniverse-mobile-dev` project.
From there, go to the Cloud Messaging tab on the left under 'Growth'
From there, select 'New Message'
Type in your message
Select which app you want to target (Android or iOS)
Send the message.

A couple of things to make sure of:
 - You need to build to an actual device
 - You need to be outside of the app to receive the notification

**Comment by noahmalmed** [comment]
@wgranger cleaned up the commented out code. Added some more useful comments about completing firebase migration

**Comment by wgranger** [comment]
Looks good. I was able to push a sample notification on an actual device. Also, several functions have commented out content, but those functions will be built out more as push notifications are fleshed out.

**Pull request merged**